modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/QueryTaskExecutorImpl.java

```diff
@@ -81,15 +81,30 @@ public void execute(UUID qryId, long fragmentId, Runnable qryTask) {
                 () -> {
                     try {
                         qryTask.run();
+                    } catch (IgniteException | IgniteInternalException iex) {
+                        //it's normal situation when we have our own exception here. Just ignore.
+                        LOG.debug("Uncaught exception", iex);
```

Comment:
instead of deciding for caller which exceptions are severe and which are not, why don't force caller to do exception handling in submitted task? In my opinion, every exception is severe on this level, thus I would just submit everything to `failureProcessor`

Reply:
agree, thanks for the note